neqo-http3/src/priority.rs

```diff
@@ -37,6 +38,30 @@ impl Priority {
             other => Some(Header::new("priority", format!("{}", other))),
         }
     }
+
+    pub fn from_bytes(bytes: &[u8]) -> Result<Priority, &'static str> {
+        let dict = Parser::parse_dictionary(bytes)?;
+        let urgency = match dict.get("u") {
+            Some(ListEntry::Item(Item {
+                bare_item: BareItem::Integer(u),
+                ..
+            })) if (0i64..=7).contains(u) => *u as u8,
+            None => 3,
+            _ => return Err("priority: invalid urgency value"),
```

Comment:
please return Error::HttpFrame.

If you want a more specific error, you can add a new one, but at the end this will be mapped into Error::HttpFrame.